handler.go

```diff
@@ -319,10 +319,16 @@ func useCache(theURL *url.URL) bool {
 
 func buildHealthcheckJSONResponse(w http.ResponseWriter, healthResult fthealth.HealthResult) {
 	w.Header().Set("Content-Type", "application/json")
-	enc := json.NewEncoder(w)
-	err := enc.Encode(healthResult)
+	jData, err := json.Marshal(healthResult)
 	if err != nil {
-		panic("Couldn't encode health results to ResponseWriter.")
+		log.WithError(err).Error("Marshaling healthResult failed.")
+		w.WriteHeader(http.StatusInternalServerError)
+		w.Write([]byte("{\"message\": \"Internal server error\"}"))
+		return
+	}
+	c, err := w.Write(jData)
+	if err != nil {
+		log.WithError(err).Error("Writing to ResponseWriter failed code=%s", c)
```

Comment:
In order to use the format string we need to use the Errorf() function, not the Error() one. Another thing to note here is that (I think) the variable **c** will not contain response code, but the number of bytes written and that is something we might just skip logging.